sklearn/tree/_tree.pyx

```diff
@@ -896,6 +1023,56 @@ cdef class BaseTree:
 
         return node_id
 
+    cdef SIZE_t _update_node(
+        self,
+        SIZE_t parent,
+        bint is_left,
+        bint is_leaf,
+        SplitRecord* split_node,
+        double impurity,
+        SIZE_t n_node_samples,
+        double weighted_n_node_samples,
+        unsigned char missing_go_to_left
+    ) except -1 nogil:
+        """Update a node on the tree.
```

Comment:
To make this work for oblique splits in scikit-tree, I guess we would need to also update `proj_vec_indices` and `proj_vec_weights`(?) Nothing to do here I think(?)

Reply:
Are those variables stored with the node?

Reply:
Yeah in downstream oblique trees, we additionally handle a projection vector of indices and weights.

To enable OF/MORF to have the partial_fit via `update` function, we will just need to perhaps override the `update_node` function to handle additionally the `proj_vec_indices/weights`. Then the `TreeBuilder.update` function can be used for any tree.